### PR TITLE
mrc-1864: Delint

### PR DESCRIPTION
**Pull request by richfitz**
A bit of spring cleaning, as enough style things are bugging me as I look through this.

The "nolint" tags are mostly to do with this unsatisfiable condition with `alist()`: https://github.com/jimhester/lintr/issues/540, but also a couple of wanted usages of `1:length(x)`

### CODE REVIEW

File: tests/testthat/examples/array_odin_user3.R [hunk 1]
```diff
@@ -3,21 +3,13 @@ dim(age_width) <- user()
...
 den[2:N_age] <- age_rate[i - 1] * den[i - 1] / (age_rate[i] + b)
 
-## test to make sure densities add up to 1
-##   TODO: should do an assert in here?
-##   TODO: sum not yet supported
-## den_tot <- sum(den)
-
 initial(S[1:N_age]) <- den[i] * (N - I0)
 initial(I[1:N_age]) <- den[i] * I0
 initial(R[1:N_age]) <- 0
```
Comment (weshinsley) [inline tests/testthat/examples/array_odin_user3.R:15]: Don't know if seq_len is helpful on orig lines 9 and 21-23?

**Review by weshinsley — APPROVED** [review]
A thriller! A couple of seq_lens to consider...

File: R/ir_parse_arrays.R [hunk 2]
```diff
@@ -528,12 +528,12 @@ ir_parse_expr_rhs_expression_sum <- function(rhs, line, source) {
...
         ## directly with empty objects
-        index <- as.list(target[-(1:2)])
+        index <- as.list(target[- (1:2)])
         target <- target[[2L]]
-        is_empty <- vlapply(index, identical, quote(expr = ))
+        is_empty <- vlapply(index, identical, quote(expr = )) # nolint
         if (any(is_empty)) {
           if (length(index) == 1L) {
-            index[] <- list(bquote(1:length(.(target))))
+            index[] <- list(bquote(1:length(.(target)))) # nolint
```
[the remark this answers is not in the document]
Comment (richfitz) [reply R/ir_parse_arrays.R:536]: This actually substitutes in the `1:n` syntax that odin uses, it's not an R function in this context. So when the user writes

```
x[] <- i
```

it is syntactic sugar for 

```
x[1:length(x)] <- i
```

which becomes, in the C:

```
for (int i = 1; i <= internal->x_length; ++i) {
  x[i - 1] <- i
}
```

File: R/ir_parse.R [hunk 3]
```diff
@@ -687,19 +687,19 @@ ir_parse_expr_lhs_index <- function(lhs, line, source) {
     ir_parse_error("array lhs must be a name", line, source)
   }
 
-  index <- as.list(lhs[-(1:2)])
+  index <- as.list(lhs[- (1:2)])
```
[the remark this answers is not in the document]
Comment (richfitz) [reply R/ir_parse.R:690]: yeah, this looks pretty weird I think

**Pull request merged**